src/pages/home/report/ReportActionCompose/SuggestionMention.js

```diff
@@ -188,17 +188,17 @@ function SuggestionMention({
             }
 
             const valueAfterTheCursor = value.substring(selectionEnd);
-            const indexOfFirstWhitespaceCharOrEmojiAfterTheCursor = valueAfterTheCursor.search(CONST.REGEX.NEW_LINE_OR_WHITE_SPACE_OR_EMOJI);
+            const indexOfFirstSpecialCharOrEmojiAfterTheCursor = valueAfterTheCursor.search(CONST.REGEX.MENTION_BREAKER);
 
-            let indexOfLastNonWhitespaceCharAfterTheCursor;
-            if (indexOfFirstWhitespaceCharOrEmojiAfterTheCursor === -1) {
-                // we didn't find a whitespace/emoji after the cursor, so we will use the entire string
-                indexOfLastNonWhitespaceCharAfterTheCursor = value.length;
+            let suggestionEndIndex;
+            if (indexOfFirstSpecialCharOrEmojiAfterTheCursor === -1) {
+                // We didn't find a special char/whitespace/emoji after the cursor, so we will use the entire string
+                suggestionEndIndex = value.length;
             } else {
-                indexOfLastNonWhitespaceCharAfterTheCursor = indexOfFirstWhitespaceCharOrEmojiAfterTheCursor + selectionEnd;
+                suggestionEndIndex = indexOfFirstSpecialCharOrEmojiAfterTheCursor + selectionEnd;
             }
 
-            const leftString = value.substring(0, indexOfLastNonWhitespaceCharAfterTheCursor);
+            const leftString = value.substring(0, suggestionEndIndex);
```

Comment:
This caused a regression when pressing the left/right arrow keys after typing '@', the text selection does not move, https://github.com/Expensify/App/issues/39102

Reply:
Hi @paultsimura, thanks for your comment. I commented here to complete the checklist (the one before closing out an issue), so the linked issue has been fixed in production. 

I believe that issue was introduced after this PR. Can you take a look at your recording -`MacOS: Chrome / Safari`?

Reply:
That's unexpected behavior as we want highlighted text to be updated when moving cursor

Reply:
See https://github.com/Expensify/App/issues/39102#issuecomment-2026542629